File: Solutions/1446_Consecutive_Characters/1446_Consecutive_Characters.py
#What you need to do in this problem is just iterate over string and find groups of equal symbols, and then return the length of the longest group. Natural way to do it is to use functionality of itertools library, more precisely groupby function. By definition, if we do not specify arguments for groupby, it will create groups exactly with equal elements, so what we need to do is just to choose group with the biggest size and return it!

#Complexity: time and space complexity is O(n), because we iterate over our data once, but also we need to have space to keep our groupby object.

class Solution:
    def maxPower(self, s: str) -> int:
         return max(len(list(j)) for _,j in groupby(s))

#Third fastest
class Solution:
    def maxPower(self, s: str) -> int:
        curr=''
        count=0
        mx=0
        for i in s:
            if i == curr:
                count+=1
            else:
                curr = i
                if count > mx:
                    mx = count
                count=0
                
        if count > mx:
                mx = count
        return mx + 1

#Second fastest
class Solution:
    def maxPower(self, s: str) -> int:
        curr=''
        count=0
        mx=0
        for i in s:
            if i == curr:
                count+=1
            else:
                curr = i
                if count > mx:
                    mx = count
                count=0
                
        if count > mx:
                mx = count
        return mx + 1

#Fastest
class Solution:
    def maxPower(self, s):
        s = chain(s, '\n')
        res = cur = 1
        prev = next(s)
        for x in s:
            if x != prev:
                if cur > res:
                    res = cur
                cur = 1
            else:
                cur += 1
            prev = x
        return res

File: Solutions/1446_Consecutive_Characters/1446_Consecutive_Characters.c
//Second fastest
int maxPower(char * s)
{
    int i,max=0,cnt=0;
    
    for(i=0;i<strlen(s);i++)
    {
        if(s[i]==s[i+1])
        {
            cnt++;
            if(max<cnt)
                max=cnt;
        }
        else
        {
            cnt=0;
        }
    }
    return(max+1);

}

//Fastest
int maxPower(char* s){
	int max = 1;
	for(int i = 0;  s[i] != '\0'; ){
	
		int count = 0;  
		int j = i;  

		while(s[i] == s[j]) count++,j++;	
		
		i = j;
		if(count > max) max = count;
		
	}
	return max;
}

//Another fastest
int maxPower(char * s){
int max=1,prev=1;
for(int i=0;i<strlen(s)-1;i++)
{
    if(s[i]==s[i+1])
    {
        max++;
    }
    else
    {
        if(max>prev)
        {
            prev=max;
        }
         max=1;
    }
}
if(max>prev)
    return max;
return prev;
}